### PR TITLE
Merge `AdminServiceEventStore` into `AdminServiceStore`

### PR DESCRIPTION
**Pull request by shannynalayna**
This PR merges the `AdminServiceEventStore` into the `AdminServiceStore`. This removes the memory-backed implementation of the `AdminServiceEventStore` and merges the methods for this trait into the `AdminServiceStore` trait. 

This PR also adds the Diesel implementations to the `AdminServiceStore` `diesel` module and includes tests for the newly implemented operations. (These tests were moved from the `AdminServiceEventStore` `diesel` module.)

This change also addresses the update where the `AdminServiceEventStore` was used, mainly in the admin service. This PR also updates the shared admin service to update the feature-guarded `admin_event_store` to take the `AdminServiceStore` trait to allow for different backend configurations for the admin store and the event store. This will mainly be necessary if the `YAML` storage option is chosen (in which case the `YamlAdminServiceStore` will be used for the admin store, and the SQLite-backed admin store will be used for the event store).

### CODE REVIEW

File: libsplinter/src/admin/store/event.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
+// Copyright 2018-2020 Cargill Incorporated
```
Comment (agunde406) [inline libsplinter/src/admin/store/event.rs:1]: The copyright in this PR should be updated to 2021

File: libsplinter/src/admin/store/event.rs [hunk 2]
```diff
@@ -53,6 +53,97 @@ impl AdminServiceEvent {
     }
 }
 
+/// Builder to be used to build a `Circuit`
```
Comment (agunde406) [inline libsplinter/src/admin/store/event.rs:56]: `Circuit` -> `AdminServiceEvent`

File: libsplinter/src/admin/rest_api/actix/ws_register_type.rs [hunk 3]
```diff
@@ -26,7 +26,7 @@ use crate::admin::service::{
     AdminCommands, AdminServiceEventSubscriber, AdminServiceStatus, AdminSubscriberError,
 };
 #[cfg(feature = "admin-service-event-store")]
-use crate::admin::store::events;
```
Comment (agunde406) [inline libsplinter/src/admin/rest_api/actix/ws_register_type.rs:29]: Typo in commit message `This includes a removing `

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: libsplinter/src/admin/store/event.rs (hunk 1 above)
Comment (shannynalayna) [reply libsplinter/src/admin/store/event.rs]: Updated

File: libsplinter/src/admin/store/event.rs (hunk 2 above)
Comment (shannynalayna) [reply libsplinter/src/admin/store/event.rs]: Updated

File: libsplinter/src/admin/rest_api/actix/ws_register_type.rs (hunk 3 above)
Comment (shannynalayna) [reply libsplinter/src/admin/rest_api/actix/ws_register_type.rs:29]: Ended up re-wording this commit after updating the admin service to still take an "event_store" (that uses the admin store's trait) so I believe this sentence/typo was removed entirely.

File: libsplinter/src/admin/store/diesel/operations/add_event.rs [hunk 4]
```diff
@@ -0,0 +1,228 @@
+// Copyright 2018-2020 Cargill Incorporated
```
Comment (agunde406) [inline libsplinter/src/admin/store/diesel/operations/add_event.rs:1]: Missed 2021 update

File: libsplinter/src/admin/store/diesel/operations/list_events.rs [hunk 5]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2018-2021 Cargill Incorporated
+// Copyright 2018-2020 Cargill Incorporated
```
Comment (agunde406) [inline libsplinter/src/admin/store/diesel/operations/list_events.rs:1]: Missed 2021 update

File: libsplinter/src/admin/store/diesel/operations/list_events_since.rs [hunk 6]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2018-2021 Cargill Incorporated
+// Copyright 2018-2020 Cargill Incorporated
```
Comment (agunde406) [inline libsplinter/src/admin/store/diesel/operations/list_events_since.rs:1]: Missed 2021 update

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: splinterd/src/daemon.rs [hunk 7]
```diff
@@ -510,8 +526,8 @@ impl SplinterDaemon {
             Box::new(AllowAllKeyPermissionManager),
             Some(self.admin_timeout),
             routing_writer.clone(),
-            #[cfg(feature = "admin-service-event-store")]
-            admin_event_store,
+            #[cfg(feature = "service-arg-validation")]
```
Comment (agunde406) [inline splinterd/src/daemon.rs:529]: This should be admin-service-event-store

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: libsplinter/src/admin/store/event.rs [hunk 8]
```diff
@@ -53,6 +53,97 @@ impl AdminServiceEvent {
...
+}
+
+impl AdminServiceEventBuilder {
+    /// Creates a new `AdminServiceEventBuilder`
+    pub fn new() -> Self {
+        AdminServiceEventBuilder::default()
+    }
+
+    /// Returns the event ID in the builder
+    pub fn event_id(&self) -> Option<i64> {
```
[the remark this answers is not in the document]
Comment (shannynalayna) [reply libsplinter/src/admin/store/event.rs:71]: The `AdminServiceEvent` struct defined in this module. These fields are used when handling the message. The event fields are also used within the diesel operations where the events are sorted by this field and filtered by this field.

File: libsplinter/src/admin/store/diesel/operations/add_event.rs (hunk 4 above)
Comment (shannynalayna) [reply libsplinter/src/admin/store/diesel/operations/add_event.rs]: Updated the copyrights from the added operations files - encompassing your other comments

File: splinterd/src/daemon.rs (hunk 7 above)
Comment (shannynalayna) [reply splinterd/src/daemon.rs]: Fixed

File: splinterd/src/daemon.rs [hunk 9]
```diff
@@ -244,6 +239,27 @@ impl SplinterDaemon {
             }
         };
 
+        #[cfg(feature = "admin-service-event-store")]
+        let event_store = {
+            if self.storage_type.is_some() {
```
Comment (agunde406) [inline splinterd/src/daemon.rs:244]: This should not default to memory if storage_type is some. It should still be whatever the database value is even if the admin service store gets set as YamlAdminServiceStore.

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: libsplinter/src/admin/store/event.rs (hunk 8 above)
[the remark this answers is not in the document]
Comment (peterschwarz) [reply libsplinter/src/admin/store/event.rs:71]: The intent of my question has more to do with why their are getters on the builder vs the event itself.

[the remark this answers is not in the document]
Comment (shannynalayna) [reply libsplinter/src/admin/store/event.rs]: Ah, yeah those weren't used for anything so removed them from that commit.

File: splinterd/src/daemon.rs (hunk 9 above)
Comment (shannynalayna) [reply splinterd/src/daemon.rs]: Updated to take the `db_url` if the storage type is yaml, and then use `memory` for the `db_url` if the storage type is memory.

**Review by peterschwarz — APPROVED** [review]

**Review by agunde406 — APPROVED** [review]

**Pull request merged**